app/src/main/java/com/duckduckgo/app/browser/httpauth/WebViewHttpAuthStore.kt

```diff
@@ -0,0 +1,128 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.httpauth
+
+import android.os.Build
+import android.webkit.WebView
+import android.webkit.WebViewDatabase
+import androidx.annotation.UiThread
+import androidx.annotation.VisibleForTesting
+import com.duckduckgo.app.browser.httpauth.db.HttpAuthDao
+import com.duckduckgo.app.browser.httpauth.db.HttpAuthEntity
+import com.duckduckgo.app.fire.fireproofwebsite.data.FireproofWebsiteDao
+import com.duckduckgo.app.fire.fireproofwebsite.data.website
+import com.duckduckgo.app.global.DispatcherProvider
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+
+data class WebViewHttpAuthCredentials(val username: String, val password: String)
+
+@VisibleForTesting
+fun WebView.setHttpAuthUsernamePasswordCompat(host: String, realm: String, username: String, password: String) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        val webViewDatabase = WebViewDatabase.getInstance(this.context)
+        webViewDatabase.setHttpAuthUsernamePassword(host, realm, username, password)
+    } else {
+        @Suppress("DEPRECATION")
+        setHttpAuthUsernamePassword(host, realm, username, password)
+    }
+
+}
+
+@VisibleForTesting
+fun WebView.getHttpAuthUsernamePasswordCompat(host: String, realm: String): WebViewHttpAuthCredentials? {
+    val credentials = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        val webViewDatabase = WebViewDatabase.getInstance(this.context)
+        webViewDatabase.getHttpAuthUsernamePassword(host, realm)
+    } else {
+        @Suppress("DEPRECATION")
+        getHttpAuthUsernamePassword(host, realm)
+    } ?: return null
+
+    return WebViewHttpAuthCredentials(
+        username = credentials[0],
+        password = credentials[1],
+    )
+}
+
+@VisibleForTesting
+fun WebView.clearAuthentication() {
+    val webViewDatabase = WebViewDatabase.getInstance(this.context)
+    webViewDatabase.clearHttpAuthUsernamePassword()
+}
+
+// Methods are marked to run in the UiThread because it is the thread of webview
+// if necessary the method impls will change thread to access the http auth dao
+interface WebViewHttpAuthStore {
+    @UiThread
+    fun setHttpAuthUsernamePassword(webView: WebView, host: String, realm: String, username: String, password: String)
+    @UiThread
+    fun getHttpAuthUsernamePassword(webView: WebView, host: String, realm: String): WebViewHttpAuthCredentials?
+    @UiThread
+    fun clearHttpAuthUsernamePassword(webView: WebView)
+}
+
+class RealWebViewHttpAuthStore(
+    private val dispatcherProvider: DispatcherProvider,
+    private val fireproofWebsiteDao: FireproofWebsiteDao,
+    private val httpAuthDao: HttpAuthDao? = null
```

Comment:
is this default param required? I like that we have the logic inside the `DatabaseModule` to decide why it's `null`, and I would expect the same from any place that creates an instance of this class.

Reply:
I can remove the `=null` yep